File: Python/7py 3/row_weights.py
"""
|--------------------------------------------------------------------------
| Row Weights
|--------------------------------------------------------------------------
|
"""

def row_weights(array):
   res = [0, 0]
   for i, v in enumerate(array):
      if i % 2 == 0: res[0] += v
      else: res[1] += v
   return tuple(res)


def row_weights(array):
   return sum(array[::2]), sum(array[1::2])


def row_weights(array):
   res = [0, 0]
   i = 0
   while i < len(array):        
      res[i % 2] += array[i]
      i += 1
   return tuple(res)


print(row_weights([80]), (80,0))
print(row_weights([100,50]), (100,50))
print(row_weights([50,60,70,80]), (120,140))
print(row_weights([13,27,49]), (62,27))

File: PHP/7php 2/rowWeights.php
<?php
/*
|--------------------------------------------------------------------------
| Row Weights
|--------------------------------------------------------------------------
|
*/

function rowWeights($arr) {
   $res = [0, 0];
   for ($i = 0; $i < count($arr); $i++) {
      $i % 2 == 0 ? ($res[0] += $arr[$i]) : ($res[1] += $arr[$i]);
   }
   return $res;
}


function rowWeights($arr) {
   $r = [0, 0];
   foreach ($arr as $k => $v) $r[$k % 2 != 0] += $v;
   return $r;
}


function rowWeights($arr) {
   return [
      array_sum(array_filter($arr, function ($v){ return !($v%2); }, ARRAY_FILTER_USE_KEY)),
      array_sum(array_filter($arr, function ($v){ return $v%2; }, ARRAY_FILTER_USE_KEY))
   ];
}


print_r(rowWeights([80]));
print_r(rowWeights([100, 50]));
print_r(rowWeights([50, 60, 70, 80]));
print_r(rowWeights([13, 27, 49]));